var/spack/repos/builtin/packages/mlhka/package.py

```diff
@@ -0,0 +1,47 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import subprocess
+
+
+class Mlhka(Package):
+    """A maximum likelihood ratio test of natural selection, using polymorphism
+       and divergence data."""
+
+    homepage = "https://wright.eeb.utoronto.ca"
+    url      = "https://github.com/rossibarra/MLHKA"
+
+    version('2.1', git='https://github.com/rossibarra/MLHKA.git',
+            commit='e735ddd39073af58da21b00b27dea203736e5467')
+
+    def patch(self):
+        command_line = (spack_cxx, 'MLHKA_version{0}.cpp'.format(self.version),
+                        '-o', 'MLHKA')
+        subprocess.Popen(command_line)
```

Comment:
In a `Package`, this stuff should go in install -- `patch()` isn't supposed to actually build anything (just prepare the source).  Can you move it?

This is probably more idiomatic:

```python
cxx = which('c++')
cxx('MLHKA_version{0}.cpp'.format(self.version), '-o', 'MLHKA')
```

The spack compiler wrappers are available *both* as specifically named things (e.g., `gcc`, `icc`, depending on what you're using), *and* their generic names (for situations like this, where the build system or lack thereof is not sensitive to the compiler name, and you just know you need the `cc` or `c++` compiler.